index.js

```diff
@@ -20,7 +47,60 @@
  * @returns boolean, valid or invalid
  */
 export function validate(str, vars) {
+  // Operator Validation
   var len = str.length;
+  var op = "";
+  // Loop to identify operator strings
+  for (var j = 0; j < len; j++) {
+    // If operator char, append to op
+    if (!(/[a-zA-Z]|[0-9]|[.]|[-]|[(]|[)]|[\s]|[,]/.test(str[j]))) {
+      op = op + str[j];
+      if (j == len - 1 || j == 0) {
+        return false;
+      }
+    }
+    // Checks operator if length > 1 and operator string ends
+    else if (op.length > 0) {
+      if (!(ops.includes(op))) {
+        return false;
+      }
+      // Validates expression to left of operator
+      var lsearch = true;
+      var op_l = j - op.length - 1;
+      while(lsearch) {
+        if (!(/[\s]/.test(str[op_l]))) {
+          if (!(/[a-zA-Z]|[0-9]|[.]|[-]|[)]|[,]/.test(str[op_l]))) {
+            return false;
+          }
+          lsearch = false;
+        } else {
+          op_l -= 1;
+        }
+      }
+      // Validates expression to right of operator
+      var rsearch = true;
+      var op_r = j;
+      if (op_r > len - 1) {
+        return false;
+      }
+      while(rsearch) {
+        if (!(/[\s]/.test(str[op_r]))) {
+          if (!(/[a-zA-Z]|[0-9]|[.]|[-]|[(]|[,]/.test(str[op_r]))) {
+            return false;
+          }
+          rsearch = false;
+        } else {
+          op_r += 1;
+        }
+      }
+      op = "";
+    }
+    else {
+      op = "";
+    }
+  }
+
+  // Function validation
   let funs = [
```

Comment:
Like your operator array, this could also be defined outside the function.

Reply:
Fixed